daft/datatype.py

```diff
@@ -32,6 +33,43 @@
         _RAY_DATA_EXTENSIONS_AVAILABLE = False
 
 
+class ImageMode(str, Enum):
```

Comment:
I opted for reimplementing it in Python rather than expose it via the pyo3 enum for the sake of discoverability and documentation, since this enum is where we'd document the available image modes and their semantics. If I'm a user and I see that the image type takes a `mode` arg, I'll want to look up the available options; we should ideally have a generated API docs page for the `ImageMode` enum that they can jump to from the `DataType` API docs page, and they should be able to jump to the `ImageMode` enum definition in Python in-code.

Is there a way to have a good discoverability/docs story for a Rust-defined enum exposed to Python via pyo3? I wasn't able to find one, and it looks like Polars did a similar duplication for e.g. `DataType` possibly for similar reasons (although their Python-side enum differs slightly from the Rust-side enum).